docker/docker_client.go

```diff
@@ -479,12 +480,13 @@ func (c *dockerClient) getBearerToken(ctx context.Context, realm, service, scope
 	default:
 		return nil, errors.Errorf("unexpected http code: %d, URL: %s", res.StatusCode, authReq.URL)
 	}
-	tokenBlob, err := ioutil.ReadAll(res.Body)
+	tokenBlob := new(bytes.Buffer)
+	_, err := tokenBlob.ReadFrom(res.Body)
 	if err != nil {
 		return nil, err
 	}
 
-	return newBearerTokenFromJSONBlob(tokenBlob)
+	return newBearerTokenFromJSONBlob(tokenBlob.Bytes())
```

Comment:
What does this change do? `ioutil.ReadAll` internally uses a `bytes.Buffer` in a very similar way.

Reply:
@mtrmac it was breaking and throwing an error "unknown end of JSON input" and when I changed it to read in the token like this, it stopped throwing the error.

Reply:
That’s surprising.  Can you compare the two values passed to `newBearerTokenFromJSONBlob`?